torch/nativert/executor/Placement.h

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include <c10/core/Device.h>
+#include <c10/util/Logging.h>
+
+#include <optional>
+#include <unordered_map>
+
+namespace torch::nativert {
+
+/**
+ * @brief Normalizes a c10::Device by ensuring a canonical form.
+ *
+ * This function returns a normalized version of the input device:
+ * - For CPU devices, the returned device will have no index (i.e., the default CPU device).
+ * - For CUDA devices, if no index is specified, index 0 is assumed.
+ * - For other device types, the function will raise an error.
+ *
+ * @param device The input c10::Device to normalize.
+ * @return A normalized c10::Device with standardized indexing.
+ *
+ * @throws c10::Error If the device type is not CPU or CUDA.
+ */
+
+TORCH_API c10::Device normalizeDevice(const c10::Device& device);
+
+/**
+ * Returns true if the two devices are the same and has the same device index (if cuda).
```

Comment:
This is only with the above definition of device index not being important on CPU?

Reply:
yeah. and the device index is not compared in the implementation if the device is cpu.

Reply:
I would be curious why you had to build so much scaffolding to special case cpu here and not just use whatever index is used there (which is going to be 0 in ~100% of the cases).

Reply:
> I would be curious why you had to build so much scaffolding to special case cpu here and not just use whatever index is used there (which is going to be 0 in ~100% of the cases).

good question, I'm not sure, as this is mostly exiting code moved from internal. @SherlockNoMad Do you know why we need to special case for CPU here?